be/src/service/service_be/lake_service.cpp

```diff
@@ -478,28 +478,38 @@ void LakeServiceImpl::get_tablet_stats(::google::protobuf::RpcController* contro
                                        const ::starrocks::lake::TabletStatRequest* request,
                                        ::starrocks::lake::TabletStatResponse* response,
                                        ::google::protobuf::Closure* done) {
+    static constexpr int64_t kDefaultTimeout = 5 * 60 * 1000L; // 5 minutes
     brpc::ClosureGuard guard(done);
     auto cntl = static_cast<brpc::Controller*>(controller);
 
     if (request->tablet_infos_size() == 0) {
         cntl->SetFailed("missing tablet_infos");
         return;
     }
-
+    int64_t timeout_ms = request->has_timeout_ms() ? request->timeout_ms() : kDefaultTimeout;
+    int64_t due_time = butil::gettimeofday_ms() + timeout_ms;
     auto thread_pool = get_tablet_stats_thread_pool(_env);
     auto latch = BThreadCountDownLatch(request->tablet_infos_size());
     bthread::Mutex response_mtx;
     for (const auto& tablet_info : request->tablet_infos()) {
         auto task = [&, tablet_info]() {
             DeferOp defer([&] { latch.count_down(); });
+
```

Comment:
does this for loop generates a task for each tablet info?

does it means that the latch will be count down repeatedly if the request contains multiple tablets?

Reply:
Yes

Reply:
ok